### PR TITLE
#163connect frontend and backend

### PR DESCRIPTION
**Pull request by BelousSofiya**
Add CORS-headers library to backend
Add axios to Frontend
Add an idea to Header in App.js
Give examples of axios requests at the beggining of LoginContent.jsx, ProfileContent.js, Companies.js, SignUpFormContent.js

Open pgadmin
Open frontend
Don't forget to make user is_active=true in db!
Reload frontend 
Go to profile page
look at network, sessionStorage and console during work

### CODE REVIEW

File: FrontEnd/public/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" />
     <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" />
     <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" />
-    <link rel="manifest" href="/site.webmanifest" />
+<!--    <link rel="manifest" href="/site.webmanifest" />-->
```
Comment (popovycholeg) [inline FrontEnd/public/index.html:9]: Are we going to implement PWA? If no we can remove it

File: FrontEnd/src/App.js
Comment (popovycholeg) [inline FrontEnd/src/App.js:1]: Remove comments from the top of file

File: FrontEnd/src/App.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ function App() {
   return (
     <BrowserRouter>
       <div className="App">
-        <Header isAuthorized={true}></Header>
+        <Header isAuthorized={sessionStorage.getItem("isAuthenticated")}></Header>
```
Comment (popovycholeg) [inline FrontEnd/src/App.js:26]: I would use redux or react context for this. This property could stay same after login. Could you check it?

Comment (BelousSofiya) [reply FrontEnd/src/App.js:26]: My teammate Ilya will work with this task. It's just an idea-proposal for him. Should i remove it?

File: FrontEnd/src/components/ProfilePage/ProfilePageComponents/ProfileContent.js [hunk 3]
```diff
@@ -10,6 +10,22 @@ import StartupInfo from '../FormComponents/StartupInfo';
...
+        'Authorization': token}})
+  .then(function (response) {
+    console.log(response.data);
+    console.log(response.status);
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+}
+
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/ProfilePageComponents/ProfileContent.js:28]: I would highly recommend using redux-thunk for managing asynchronous requests. You may utilize it in pair with [redux-toolkit](https://redux-toolkit.js.org/introduction/getting-started) to simplify work lift. If the team doesn't have enough time to learn redux I would recommend using [react-query](https://tanstack.com/query/latest/docs/react/overview) for managing API requests which is easier to start

File: FrontEnd/src/components/SignUp/components/signup/signup-form/SignUpFormContent.js [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
+        re_password: 'pASSWord123'},
+        {withCredentials: true}
+        )
+      .then(function (response) {
+        console.log(response.data);
+        console.log(response.status);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
```
Comment (popovycholeg) [inline FrontEnd/src/components/SignUp/components/signup/signup-form/SignUpFormContent.js:25]: Same here we do nothing with data that we fetched. Consider using react-query. If you do just query in the components it should be done at least in `useEffect` hook

File: FrontEnd/src/components/authorization/LoginContent.jsx [hunk 5]
```diff
@@ -1,10 +1,25 @@
 import { useForm } from "react-hook-form";
 import { useState, useEffect} from "react";
+import axios from 'axios';
 import validator from "validator";
 import EyeVisible from "./EyeVisible";
 import EyeInvisible from "./EyeInvisible";
 import classes from "./LoginContent.module.css";
 
+axios
```
Comment (popovycholeg) [inline FrontEnd/src/components/authorization/LoginContent.jsx:9]: I think react-query could be the easiest option for us. You may follow [this tutorial ](https://dev.to/this-is-learning/react-query-authentication-flow-id2)

File: FrontEnd/public/index.html (hunk 1 above)
Comment (BelousSofiya) [reply FrontEnd/public/index.html]: Thank you for feedback. Done.

File: FrontEnd/src/App.js
Comment (BelousSofiya) [reply FrontEnd/src/App.js:1]: Thank you for feedback. Done.

File: FrontEnd/src/App.js (hunk 2 above)
Comment (BelousSofiya) [reply FrontEnd/src/App.js]: Thank you. Done.

File: FrontEnd/src/components/ProfilePage/ProfilePageComponents/ProfileContent.js (hunk 3 above)
Comment (BelousSofiya) [reply FrontEnd/src/components/ProfilePage/ProfilePageComponents/ProfileContent.js:28]: Thank you for feedback. My task was to connect using axios, to write little examples of requests for teammates. I'll tell about your advice to use react-query to my teammates.

File: forum/settings.py [hunk 6]
```diff
@@ -54,7 +55,14 @@
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    'corsheaders.middleware.CorsMiddleware',
 ]
+CORS_ALLOW_CREDENTIALS = True
+CORS_ORIGIN_WHITELIST = (
+    'http://localhost:3000',
+    '0.0.0.0',
```
Comment (ProDG) [inline forum/settings.py:63]: Having this, all other entries are redundant. Though, it's very unsafe.

File: profiles/views.py [hunk 7]
```diff
@@ -127,3 +127,14 @@ class ViewedCompanyList(ListCreateAPIView):
...
+
+class GetUserId(ListAPIView):
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def list(self, request):
+        profile = Profile.objects.get(person_id=self.request.user)
+        serializer = ProfileSerializer(profile)
+        return Response(serializer.data)
```
Comment (ProDG) [inline profiles/views.py:140]: 1. Why `ListAPIView`?
2. What for do we have `queryset` property here, is it used anywhere?
3. What for do we have `searializer_class` property here?
3. Why do we _still_ (after probablly 10+ mentions in other PRs) 'handcraft' reponses?

File: profiles/views.py [hunk 8]
```diff
@@ -127,3 +127,14 @@ class ViewedCompanyList(ListCreateAPIView):
     def get_queryset(self):
         user_id = self.request.user.id
         return ViewedCompany.objects.filter(user=user_id)
+
+
+class GetUserId(ListAPIView):
```
Comment (ProDG) [inline profiles/views.py:132]: There should be NO VERBS in view names.

File: FrontEnd/src/components/landing-page/companies/Companies.js [hunk 9]
```diff
@@ -1,6 +1,14 @@
+import axios from 'axios';
 import './Companies.css';
 import {ProductCard} from "./companies-product-cards/CompaniesProductCards";
 
+const companiesList = axios.get('http://localhost:8000/api/profiles/')
+    .then(function (response) {
+    console.log(response.data);
+    console.log(response.status);
+  })
```
Comment (ProDG) [inline FrontEnd/src/components/landing-page/companies/Companies.js:9]: Formatting is terrible. 
```
const companiesList = axios.get('http://localhost:8000/api/profiles/')
    .then(function (response) {
        console.log(response.data);
        console.log(response.status);
    })
    .catch(error => console.log(error));
```

File: profiles/views.py (hunk 7 above)
Comment (ProDG) [inline profiles/views.py:140]:
``` python
class UserSelf(RetrieveAPIView):
    permission_classes = (IsAuthenticated,)
    serializer_class = ProfileSerializer

    def get_object(self):
        return self.request.user.profile
```

File: profiles/urls.py [hunk 10]
```diff
@@ -10,4 +10,5 @@
     path('saved-list/', SavedCompaniesListCreate.as_view(), name='saved_companies_list_create'),
     path('saved-list/<pk>/', SavedCompaniesDestroy.as_view(), name='saved_companies_destroy'),
     path('viewed-list/', ViewedCompanyList.as_view(), name='viewed_company_list'),
+    path('getuserid/', GetUserId.as_view(), name='get-user-id'),
```
[the remark this answers is not in the document]
Comment (ProDG) [reply profiles/urls.py:13]: I'd name it `self/` or smth like this.

File: forum/settings.py (hunk 6 above)
Comment (BelousSofiya) [reply forum/settings.py:63]: I don't understand what is redundant? CORS_ALLOW_CREDENTIALS or CORS_ORIGIN_WHITELIST? Do I have to delete something?

File: profiles/views.py (hunk 7 above)
Comment (BelousSofiya) [reply profiles/views.py]: Thank you for feedback. I have deleted it at all. I am using filtering for now

File: profiles/views.py (hunk 8 above)
Comment (BelousSofiya) [reply profiles/views.py]: Thank you for feedback. I have deleted it at all. But I'll remember about verbs

File: profiles/urls.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (BelousSofiya) [reply profiles/urls.py]: Thank you for feedback. I have deleted it at all. Just endpoint 'profiles' with query parameters

File: profiles/views.py (hunk 7 above)
Comment (BelousSofiya) [reply profiles/views.py]: Thank you for feedback. I have deleted it at all. I am using filtering for now

**Pull request merged**